File: tools/runtime-comparison/fixtures/lua/decisions.lua
local helper = require('@mado/helper')

return {
    decide = function(options, template, ocr)
        local available = {
            template = template ~= nil and template.score >= options.recognition.threshold,
            ocr = ocr ~= nil and ocr.score >= options.recognition.threshold,
        }
        return helper.choose(options.priorities, available)
    end,
}

File: tools/runtime-comparison/fixtures/javascript/decisions.js
import { choose } from '@mado/helper';

export function decide(options, template, ocr) {
    const available = {
        template: template !== null && template.score >= options.recognition.threshold,
        ocr: ocr !== null && ocr.score >= options.recognition.threshold,
    };
    return choose(options.priorities, available);
}
